Detectors/FIT/FV0/simulation/include/FV0Simulation/Digitizer.h

```diff
@@ -0,0 +1,100 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_FV0_DIGITIZER_H
+#define ALICEO2_FV0_DIGITIZER_H
+
+#include "CommonDataFormat/InteractionRecord.h"
+#include "DataFormatsFV0/Digit.h"
+#include "DataFormatsFV0/MCLabel.h"
+#include "FV0Simulation/Detector.h"
+#include "SimulationDataFormat/MCTruthContainer.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+#include "FV0Simulation/DigitizationParameters.h"
+#include <TH1F.h>
+#include "MathUtils/CachingTF1.h"
+#include "MathUtils/RandomRing.h" //added by a.khuntia
+
+namespace o2
+{
+namespace fv0
+{
+class Digitizer
+{
+ public:
+  Digitizer(const DigitizationParameters& params, Int_t mode = 0)
+    : mMode(mode),
+      parameters(params),
+      mPmtChargeVsTime(48),
+      mPmtResponse()
+  {
+    init();
+  };
+
+  ~Digitizer() = default;
+
+  void process(const std::vector<o2::fv0::Hit>* hits, o2::fv0::Digit* digit, std::vector<std::vector<double>>& channel_times);
+  void computeAverage(o2::fv0::Digit& digit);
+
+  void initParameters();
+  void printParameters();
+  void setEventTime(double value) { mEventTime = value; }
+  void setEventID(Int_t id) { mEventID = id; }
+  void setSrcID(Int_t id) { mSrcID = id; }
+  void setInteractionRecord(uint16_t bc, uint32_t orbit)
+  {
+    mIntRecord.bc = bc;
+    mIntRecord.orbit = orbit;
+  }
+  const o2::InteractionRecord& getInteractionRecord() const { return mIntRecord; }
+  o2::InteractionRecord& getInteractionRecord(o2::InteractionRecord& src) { return mIntRecord; }
+  void setInteractionRecord(const o2::InteractionRecord& src) { mIntRecord = src; }
+  uint32_t getOrbit() const { return mIntRecord.orbit; }
+  uint16_t getBC() const { return mIntRecord.bc; }
+
+  void setTriggers(o2::fv0::Digit* digit);
+  Int_t SimulateLightYield(Int_t pmt, Int_t nPhot);                 //fdd
+  Float_t SimulateTimeCfd(Int_t channel);                           //fdd
+  static Float_t PmtResponse(const Float_t* x, const Float_t* par); //fdd
```

Comment:
Since this is a well defined function, I would pass everything as spelled-out arguments.
There is no need to use pointers and this will actually prevent some compiler optimization.
(I guess this comes from the original use as a TF1)

Reply:
Done